### DISCUSSION TITLE
Is it possible to package into single whl, both main package and local sub package (with setup.py), listed as a dependency for main package?

### DISCUSSION THREAD

**Discussion by AntonOvsyannikov** [python-poetry/poetry#8556] [Q&A] [answered]
Hello.

I describe problem in details on [SO](https://stackoverflow.com/questions/77312810/is-it-possible-to-package-into-single-whl-both-main-package-and-sub-package-wi) Here is short brief.

Let's consider the following directory structure
```
+-main_package
+-sub_package
  +- setup.py
pyproject.toml
```
In `pyproject.toml`, the `sub_package` is listed as dependency
```
[tool.poetry.dependencies]
sub_package= { path = "./sub_package", develop = true }
```
Also, there is `setup.py` file in `sub_package` with some entry points, so `sub_package` should be built during `main_package` installation.

I need to build single wheel for the whole project, so I add to `pyproject.toml`
```
packages = [
    { include = "main_package" },
    { include = "sub_package" },
]
```
perform `poetry build` and got `main_package-0.0.1-p3-none-any.whl` file which looks pretty ok, but can not be installed. After looking into whl I found that `sub-package` is listed as dependency in `METADATA` file with **absolute path on my PC**, so this whl can not be installed correctly outside of my PC (but installed successfully on it)

```
Requires-Dist: sub_package @ file:///C:/Users/anton/Documents/main_package/sub_package
```

So could you please tell is it me who doing something completely wrong, or there is a way to fix it?

**Comment by radoering** [comment, marked as answer]
> After looking into whl I found that `sub-package` is listed as dependency in `METADATA` file with **absolute path on my PC**

If you want to avoid that, you should not declare it as main dependency but as dev dependency. See [dependency groups](https://python-poetry.org/docs/managing-dependencies#dependency-groups). On the other side I'm not sure if it makes sense at all to declare it as dependency when putting it in the wheel anyway.

> Also, there is setup.py file in sub_package with some entry points, so sub_package should be built during main_package installation.
> 
> I need to build single wheel for the whole project, so I add to pyproject.toml

I suppose this setup will not work because the `setup.py` in `sub_package` will be ignored if you build your project. It probably makes more sense to put all entry points (`main_package` and `sub_package`) in the `pyproject.toml` of your project.
